lessons/snake/drawing/index.md

```diff
@@ -0,0 +1,558 @@
+
+```python
+from pathlib import Path
+
+TILES_DIRECTORY = Path('snake-tiles')
+
+for path in TILES_DIRECTORY.glob('*.png'):
+    print(path)
+```
+
+My z každého souboru potřebujeme nejlépe jméno, tedy místo
+`snake-tiles/right-head.png` jenom `right-head`.
+Na to naštěstí existuje atribut `stem` (*kořen*, t.j. jméno bez přípony).
+Místo `print(path)` použij:
+
+```python
+    print(path.stem)
+```
+
+Funguje? Máš vypsané všechny možné kousků hada?
+
+Teď budeme chtít načíst obrázky do *slovníku*.
+*Klíče* slovníku, podle kterých budeme vyhledávat, budou jména která jsi
+právě vypsal{{a}}.
+*Hodnoty* pak budou Pygletí obrázky, které ve hře můžeš rovnou vykreslit.
+
+Začni s prázdným slovníkem, `{}`, a v cyklu `for` do něj postupně přidávej
+záznamy.
+Pak slovník vypiš.
+
+Až to bude mít, měl by výpis vypadat zhruba nějak takhle:
+
+```
+{'right-tongue': <ImageData 64x64>, 'top-tongue': <ImageData 64x64>,
+ 'right-top': <ImageData 64x64>, 'left-bottom': <ImageData 64x64>,
+ 'tail-left': <ImageData 64x64>, 'bottom-tongue': <ImageData 64x64>,
+ 'left-top': <ImageData 64x64>, 'bottom-bottom': <ImageData 64x64>,
+ ...
+```
+
+{% filter solution %}
+```python
+from pathlib import Path
+
+import pyglet
+
+TILES_DIRECTORY = Path('snake-tiles')
+
+snake_tiles = {}
+for path in TILES_DIRECTORY.glob('*.png'):
+    snake_tiles[path.stem] = pyglet.image.load(path)
+
+print(snake_tiles)
+```
+{% endfilter %}
+
+
+## Housenka
+
+A teď zkus načtení obrázků začlenit do programu s hadem!
+
+Všechny importy patří nahoru, konstanty pod ně, a dál pak zbytek kódu.
+Vypisovat načtený slovník ve hře nemusíš, zato místo `green_image`
+pak ve vykreslovací funkci použij třeba `snake_tiles['tail-head']`.
+
+Místo čtverečků se teď objeví kuličky – místo hada budeš mít „housenku“.
+
+{{ figure(
+    img=static('caterpillar.png'),
+    alt="Housenka",
+) }}
+
+{% filter solution %}
+```python
+from pathlib import Path
+
+import pyglet
+
+TILE_SIZE = 64
+TILES_DIRECTORY = Path('snake-tiles')
+
+snake = [(1, 2), (2, 2), (3, 2), (3, 3), (3, 4), (3, 5), (4, 5)]
+food = [(2, 0), (5, 1), (1, 4)]
+
+red_image = pyglet.image.load('apple.png')
+snake_tiles = {}
+for path in TILES_DIRECTORY.glob('*.png'):
+    snake_tiles[path.stem] = pyglet.image.load(path)
+
+window = pyglet.window.Window()
+
+@window.event
+def on_draw():
+    window.clear()
+    pyglet.gl.glEnable(pyglet.gl.GL_BLEND)
+    pyglet.gl.glBlendFunc(pyglet.gl.GL_SRC_ALPHA, pyglet.gl.GL_ONE_MINUS_SRC_ALPHA)
+    for x, y in snake:
+        snake_tiles['tail-head'].blit(
+            x * TILE_SIZE, y * TILE_SIZE, width=TILE_SIZE, height=TILE_SIZE)
+    for x, y in food:
+        red_image.blit(
+            x * TILE_SIZE, y * TILE_SIZE, width=TILE_SIZE, height=TILE_SIZE)
+
+pyglet.app.run()
+```
+{% endfilter %}
+
+
+## Jak vybrat čtverečky?
+
+Místo toho, aby byl všude stejný kousek hada,
+ale budeme chtít vybrat vždycky ten správný.
+
+Jak na to?
+
+Podle čeho ho vybrat?
+
+Vytvoř soubor `smery.py` a napiš do něj:
```

Comment:
Doteď bylo vše anglicky. Najednou do toho vstupují české názvy souborů. U toho směru, který nebude součástí výsledku to ještě beru, ale později je v souboru had střída State. Nemyslím si, že je edukativní dělat takovéhle skopičiny s mícháním angličtiny a češtiny, a IMHO je dobré se držet jednoho jazyka.

Reply:
https://github.com/pyvec/naucse.python.cz/pull/357

Reply:
Zařazeno v https://github.com/pyvec/naucse.python.cz/issues/356